src/dns/dnsConnectionView.js

```diff
@@ -0,0 +1,201 @@
+/**
+ *
+ *  Copyright 2016 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+
+/* eslint-disable no-mixed-operators */
+
+
+import * as THREE from 'three';
+import _ from 'lodash';
+
+import ConnectionView from '../base/connectionView';
+import GlobalStyles from '../globalStyles';
+
+const Console = console;
+
+function distance (a, b) {
+  return Math.sqrt(Math.pow(a.x - b.x, 2) + Math.pow(a.y - b.y, 2));
+}
+
+// Given three points, A B C, find the center of a circle that goes through all three.
+function CalculateCircleCenter (A, B, C) {
```

Comment:
Can exceed call stack limit if one of the slopes is always 0.
Ex: `CalculateCircleCenter({x:1, y:1}, {x:2, y:1}, {x:3, y:1})`

Should have a base return, or only try three times.